dockerfile/torch2.1-cuda12.1.dockerfile

```diff
@@ -0,0 +1,53 @@
+FROM nvcr.io/nvidia/pytorch:23.04-py3
+
+# Ubuntu: 20.04
+# Python: 3.8
+# CUDA: 12.1.0
+# cuDNN: 8.9.0
+# NCCL: v2.16.2-1 + FP8 Support
+# PyTorch: 2.1.0a0+fe05266f
+
+LABEL maintainer="MS-AMP"
+
+ENV DEBIAN_FRONTEND=noninteractive
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+    autoconf \
+    automake \
+    bc \
+    build-essential \
+    curl \
+    dmidecode \
+    git \
+    iproute2 \
+    jq \
+    moreutils \
+    net-tools \
+    openssh-client \
+    openssh-server \
+    sudo \
+    util-linux \
+    vim \
+    wget \
+    && \
+    apt-get autoremove && \
+    apt-get clean && \
+    rm -rf /var/lib/apt/lists/* /tmp/*
+
+ARG NUM_MAKE_JOBS=
+ENV PATH="${PATH}" \
+    LD_LIBRARY_PATH="/usr/local/lib:${LD_LIBRARY_PATH}"
+
+WORKDIR /opt/msamp
+
+ADD third_party third_party
+RUN cd third_party/nccl && \
+    make -j ${NUM_MAKE_JOBS} src.build NVCC_GENCODE="\
+    -gencode=arch=compute_70,code=sm_70 \
+    -gencode=arch=compute_80,code=sm_80 \
+    -gencode=arch=compute_90,code=sm_90" && \
+    make install
```

Comment:
Does 'make install' override the system nccl? I find the official installation step here, it create a package and install it as root.
https://github.com/NVIDIA/nccl#install

Reply:
I saw you put '/usr/local/lib" to the environment variable LD_LIBRARY_PATH. `make install` will install nccl to /usr/local/lib.  It should not be an issue.